File: packages/revcrm_accounts/src/models/Address.ts

import * as rev from 'rev-models';
import { EntityModel } from 'revcrm/lib/models/BaseModels';
import { ApiOperations } from 'rev-api/lib/decorators';
import { COUNTRIES } from './countries';

@ApiOperations(['create', 'read', 'update', 'remove'])
export class Address extends EntityModel<Address> {

    @rev.TextField({ label: 'Address Name' })  // e.g. Head Office
        name: string;
    @rev.TextField({ label: 'Address Line 1' })
        address1: string;
    @rev.TextField({ label: 'Address Line 2', required: false })
        address2?: string;
    @rev.TextField({ label: 'City' })
        city: string;
    @rev.TextField({ label: 'Region' })
        region: string;
    @rev.TextField({ label: 'Postal Code' })
        postal_code: string;
    @rev.SelectField({ label: 'Country', selection: COUNTRIES })
        country: string;

}

File: src/main/kotlin/org/revcrm/models/accounts/Address.kt
package org.revcrm.models.accounts

import org.revcrm.models.BaseModel
import javax.persistence.Entity

@Entity
class Address (
        var name: String,
        var address1: String,
        var address2: String,
        var city: String,
        var region: String,
        var postal_code: String,
        var country: String
): BaseModel()
